main/src/ca/uwaterloo/flix/language/ast/SemanticOperator.scala

```diff
@@ -805,3 +807,93 @@ object SemanticOperator {
   }
 
 }
+
+object SemanticOperatorOps {
+
+  def toUnaryOp(sop: SemanticOperator): UnaryOperator = sop match {
+    case SemanticOperator.BoolOp.Not => UnaryOperator.LogicalNot
+    case SemanticOperator.Float32Op.Neg | SemanticOperator.Float64Op.Neg
+         | SemanticOperator.Int8Op.Neg | SemanticOperator.Int16Op.Neg | SemanticOperator.Int32Op.Neg | SemanticOperator.Int64Op.Neg
+         | SemanticOperator.BigIntOp.Neg => UnaryOperator.Minus
+    case SemanticOperator.Int8Op.Not | SemanticOperator.Int16Op.Not | SemanticOperator.Int32Op.Not | SemanticOperator.Int64Op.Not
+         | SemanticOperator.BigIntOp.Not => UnaryOperator.BitwiseNegate
+    case _ => throw InternalCompilerException(s"Unexpected unary operator: '$sop'.")
```

Comment:
The match should be exhaustive, so this can be removed?

Reply:
It's not actually, that's why https://github.com/flix/flix/runs/1651149395 failed.

Reply:
```
Error:  /home/runner/work/flix/flix/main/src/ca/uwaterloo/flix/language/ast/SemanticOperator.scala:811: match may not be exhaustive.
25
It would fail on the following inputs: Add, Add, Add, Add, Add, Add, Add, And, And, And, And, And, And, Concat, Div, Div, Div, Div, Div, Div, Div, Eq, Eq, Eq, Eq, Eq, Eq, Eq, Eq, Eq, Eq, EqNull, Exp, Exp, Exp, Exp, Exp, Exp, Exp, Ge, Ge, Ge, Ge, Ge, Ge, Ge, Ge, Gt, Gt, Gt, Gt, Gt, Gt, Gt, Gt, Le, Le, Le, Le, Le, Le, Le, Le, Lt, Lt, Lt, Lt, Lt, Lt, Lt, Lt, Mul, Mul, Mul, Mul, Mul, Mul, Mul, Neq, Neq, Neq, Neq, Neq, Neq, Neq, Neq, Neq, Neq, NeqNull, Or, Or, Or, Or, Or, Or, Rem, Rem, Rem, Rem, Rem, Rem, Rem, Shl, Shl, Shl, Shl, Shl, Shr, Shr, Shr, Shr, Shr, Sub, Sub, Sub, Sub, Sub, Sub, Sub, Xor, Xor, Xor, Xor, Xor
26
Error:  /home/runner/work/flix/flix/main/src/ca/uwaterloo/flix/language/ast/SemanticOperator.scala:820: match may not be exhaustive.
27
It would fail on the following inputs: EqNull, Neg, Neg, Neg, Neg, Neg, Neg, Neg, NeqNull, Not, Not, Not, Not, Not, Not
```

Reply:
OK, I get it. Its because of the structure of SemanticOperator. Carry on and ignore my ramblings about this one.